lms/djangoapps/api_manager/users/views.py

```diff
@@ -1003,17 +1021,51 @@ def get(self, request, user_id, course_id):
             user=student,
             course_id__exact=course_key,
         )
-        current_grade = 0
-        proforma_grade = 0
-        progress_summary = {}
-        grade_summary = {}
-        grading_policy = {}
+
         if len(queryset):
-            current_grade = queryset[0].grade
-            proforma_grade = queryset[0].proforma_grade
-            progress_summary = json.loads(queryset[0].progress_summary)
-            grade_summary = json.loads(queryset[0].grade_summary)
-            grading_policy = json.loads(queryset[0].grading_policy)
+            gradebook_entry = queryset[0]
+            if (gradebook_entry.grade and
+                    gradebook_entry.proforma_grade and
+                    gradebook_entry.progress_summary and
+                    gradebook_entry.grade_summary and
+                    gradebook_entry.grading_policy):
+                current_grade = queryset[0].grade
+                proforma_grade = queryset[0].proforma_grade
+                progress_summary = json.loads(queryset[0].progress_summary)
+                grade_summary = json.loads(queryset[0].grade_summary)
+                grading_policy = json.loads(queryset[0].grading_policy)
+            else:
+                gradebook_values = _recalculate_grade(request, student, course_descriptor)
+                gradebook_entry.grade = gradebook_values["current_grade"]
+                gradebook_entry.proforma_grade = gradebook_values["proforma_grade"]
+                gradebook_entry.progress_summary = json.dumps(gradebook_values["progress_summary"], cls=EdxJSONEncoder)
+                gradebook_entry.grade_summary = json.dumps(gradebook_values["grade_summary"], cls=EdxJSONEncoder)
+                gradebook_entry.grading_policy = json.dumps(gradebook_values["grading_policy"], cls=EdxJSONEncoder)
+                gradebook_entry.save()
```

Comment:
@e-kolpakov when did you capture this traceback? Which credentials did you use and on which server?

I tried to reproduce issues described in MCKIN-3494 and MCKIN-3495 on integration server with credentials provided in the description and it seems to work without throwing 500 error.